cmd/virt-launcher/virt-launcher.go

```diff
@@ -325,6 +332,27 @@ func waitForFinalNotify(deleteNotificationSent chan watch.Event,
 	}
 }
 
+func runDownwardMetricsVirtioServer(stop chan struct{}) error {
+	if _, ok := os.LookupEnv("VIRT_LAUNCHER_DOWNWARDMETRICS_SERVER"); ok {
+		nodeName, ok := os.LookupEnv("NODE_NAME")
+		if !ok {
+			return errors.New("NODE_NAME environment variable not found")
+		}
+
+		err := virtioserial.RunDownwardMetricsVirtioServer(nodeName, stop)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func safeCloseChannel(ch chan struct{}) {
```

Comment:
Sounds like a sync issue to me if multiple code paths try to close the chan?

Reply:
Did you encounter issues with this during testing?

Reply:
No, but it can happen, it's difficult to force that situation while testing, but I think it's more robust doing that instead of just panic if that happen, leaving the guest in a possible inconsistent state (take into account that qemu already started when we try to start the server)

This is  unrelated with this PR: The is another unhandled panic in the virt-launcher, I think is better to not panic if `virtwrap.NewLibvirtDomainManager(..)` returns an error or take care of that panic and close `stopChan` channel.

Reply:
Ok, looks good to me but maybe someone else can verify.

Please open an issue for the unhandled panic.

Reply:
Ok